aws-datastore/src/androidTest/java/com/amplifyframework/datastore/HybridCloudSyncInstrumentationTest.java

```diff
@@ -0,0 +1,234 @@
+import com.amplifyframework.core.category.CategoryConfiguration;
+import com.amplifyframework.core.category.CategoryType;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.ModelSchema;
+import com.amplifyframework.core.model.query.QueryOptions;
+import com.amplifyframework.datastore.appsync.AppSyncClient;
+import com.amplifyframework.datastore.appsync.SerializedModel;
+import com.amplifyframework.datastore.appsync.SynchronousAppSync;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.logging.AndroidLoggingPlugin;
+import com.amplifyframework.logging.LogLevel;
+import com.amplifyframework.testmodels.commentsblog.AmplifyModelProvider;
+import com.amplifyframework.testmodels.commentsblog.Blog;
+import com.amplifyframework.testmodels.commentsblog.BlogOwner;
+import com.amplifyframework.testutils.HubAccumulator;
+import com.amplifyframework.testutils.Resources;
+import com.amplifyframework.testutils.sync.SynchronousApi;
+import com.amplifyframework.testutils.sync.SynchronousDataStore;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static com.amplifyframework.datastore.DataStoreHubEventFilters.publicationOf;
+import static com.amplifyframework.datastore.DataStoreHubEventFilters.receiptOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Validates the ability of the {@link AWSDataStorePlugin} to sync data up/down
+ * from AppSync, when being used by means of "Hybrid-friendly" API invocations:
+ *
+ *   1. Saving a {@link SerializedModel} via
+ *      {@link AWSDataStorePlugin#save(Model, Consumer, Consumer)};
+ *   2. Deleting a {@link SerializedModel} via
+ *      {@link AWSDataStorePlugin#delete(Model, Consumer, Consumer)};
+ *   3. Querying for {@link SerializedModel} by means of
+ *      {@link AWSDataStorePlugin#query(String, QueryOptions, Consumer, Consumer)}.
+ *
+ * This test expects a backend API that has support for the {@link Blog} family of models,
+ * which were defined by the schema in:
+ * testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/schema.graphql.
+ */
+@Ignore(
+    "Over time, this test will create a large DynamoDB table. Even if we delete the content " +
+    "through the AppSyncClient utility, the database will have lots of tombstone'd rows. " +
+    "These entries will be synced, the next time this test runs, and the DataStore initializes. " +
+    "After several runs, that sync will grow large and timeout the test, before the test can " +
+    "run any business logic. A manual workaround exists, by running this cleanup script: " +
+    "https://gist.github.com/jamesonwilliams/c76169676cb99c51d997ef0817eb9278#quikscript-to-clear-appsync-tables"
```

Comment:
May be something we can do before we run integration tests. I'll make a note of it

Reply:
Yes, I think so! I'll work with you on this. Selective Sync may also make this problem go away, if we use a TTL field of some kind.

Reply:
> Selective Sync may also make this problem go away, if we use a TTL field of some kind.

Ooh!  That's a neat idea!